dask/dataframe/shuffle.py

```diff
@@ -515,6 +516,11 @@ def rearrange_by_column_tasks(
         for i, inp in enumerate(inputs)
     }
 
+    # We know that it is _always_ beneficial to prioritize the
+    # getitem() task because it makes it possible for Dask to free
+    # the output of shuffle_group() as fast as possible.
```

Comment:
This statement sounds true of most `getitem`s. Most of the time the output will be smaller than the input. What makes it especially important for the shuffle-split workload?

Reply:
Agree, if we are _getting_ all of the parent task so that it can be freed, prioritizing `getitem()` makes most often sense.
But in the general case `getitem()` could imply any kind of memory/compute usage e.g getting an item form a [`zict`](https://github.com/dask/zict) dictionary can be very expensive.

Reply:
Right, in this case I think that the condition for this being useful is that the memory use of all of the dependents is less than or equal to the size of the dependency (equal in this case).

Reply:
If we're submitting this graph along with other graphs at the same time (maybe the shuffle is part of a larger computation) will this also prioritize the shuffle code above those other parts of the graph, or will things still operate normally?

Reply:
> If we're submitting this graph along with other graphs at the same time (maybe the shuffle is part of a larger computation) will this also prioritize the shuffle code above those other parts of the graph, or will things still operate normally?

This PR effectively makes the scheduler schedule the `getitem()` tasks and their parent task `shuffle_group()` as a single node. The priority of the shuffle code as a whole is not changed thus things should still operate normally,